### ISSUE TITLE
Docs for handing error pages are not ideal

### ISSUE THREAD

**Issue by ctolkien** [umbraco/UmbracoDocs#6420]
## What type of issue is it? (Choose one - delete the others)
Wrong documentation

## What article/section is this about?
[(link here)](https://docs.umbraco.com/umbraco-cms/tutorials/custom-error-page#configure-the-500-error-page-programmatically)

## Describe the issue

This section describes intercepting the 500 error response and then _redirecting_ the user to a page to display the error. In this scenario statuscode is lost, and the user never actually receives a 500 error, they end up receiving a 3XX redirection.

This should probably be using some combination of `app.UseExceptionHandler("...");` and `app.UseStatusCodePagesWithReExecute("/error/{0}");`

**Comment by sofietoft** [comment]
Thanks for reporting this @ctolkien ! 💪 Much appreciated!

Now, this might be a stupid question, so please forgive me, but isn't it in most cases better for the user to not see the error, than for them to see it? 😅 

In any case, would you be up for suggestion a rewrite of this guide, to include your suggestions?
Sounds like you already have some experience with this, and might know how it could be set up differently.

**Comment by ctolkien** [comment]
>Now, this might be a stupid question, so please forgive me, but isn't it in most cases better for the user to not see the error, than for them to see it?

Yep, to be clear, the issue is not the content that is shown. It is that the current flow is:

* UnhandledExceptionIsRaised -> Global Exception Handler fires -> Re-executes the /error controller -> User is sent a 302 redirect to the error page (`/statuscode/500`) -> user browser redirects to the end page, is shown the "error" page, but with a 200 ("OK") status code.

When a user hits an error, ideally the URL does not change, they should get the error on the same URL they are on, showing the error content, but at the current URL with a 500 status code.

>In any case, would you be up for suggestion a rewrite of this guide, to include your suggestions?
Sounds like you already have some experience with this, and might know how it could be set up differently.

Yes, let me do a bit more digging on handling the use cases. There are some further considerations, for instance you ideally want to have the 500 page be static content and not be rendered via the CMS. If there is an issue in your Layout file (for instance), which means that no page can work - including the 500 page, then your user will not see the properly rendered 500 page. By having it be as simple as possible, with no dependencies, that would be the most reliable option.

**Comment by sofietoft** [comment]
Aha, yes, I see what you mean 🤔 
By using the redirect method the context is lost and what they're seeing is actually "okay".

It would be great if you could suggest a better method here!
I like the idea with using a static page - that should always be available to show, at least in some form!

[Hacktoberfest](https://hacktoberfest.com/) starts next month, and this sounds like a perfect little side task for that 😄

**Comment by sofietoft** [comment]
Hi there 👋 

I'm doing some tidying up on the Issue Tracker, and since we've seen no activity on this one, I will close the issue for now.

Please feel free to re-open or reach out again.

Thanks for helping us keep the Umbraco Docs in a good place!
